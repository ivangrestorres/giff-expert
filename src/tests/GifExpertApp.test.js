import React from "react";
import "@testing-library/jest-dom";
import GifExpertApp from "../GifExpertApp";

const { shallow } = require("enzyme");

describe("Pruebas <GifExpertApp />", () => {
    test("debe mostrar el componente", () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de mostrar lista de categorias", () => {
        const categories = ["One Punch", "Mario"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });
});
