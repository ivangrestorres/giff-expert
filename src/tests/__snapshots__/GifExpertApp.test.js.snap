// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifExpertApp /> debe de mostrar lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Mario"
      key="Mario"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas <GifExpertApp /> debe mostrar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
